Add oab_broker_configuration

adellape opened the pull request:
Preview:

http://file.rdu.redhat.com/~adellape/121317/brokerconfig/install_config/oab_broker_configuration.html

PTAL @dymurray

adellape commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,590 @@
+(*ansible-service-broker* by default).
+
+.Example OpenShift Ansible Broker Configuration File
+[source,yaml]
```
Grabbed this from https://github.com/openshift/ansible-service-broker/blob/master/etc/example-config.yaml.

bmcelvee commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,599 @@
+[[oab-broker-config-registry-mock]]
+=== Mock Registry
+
+Using a mock registry is useful for reading local APB specs. Instead of going
```
s/Using a mock/A mock

adellape replied:
:+1:

bmcelvee commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,599 @@
+[[oab-broker-config-registry-dockerhub]]
+=== Dockerhub Registry
+
+Using the `dockerhub` registry allows you to load APBs from a specific
```
s/Using the `dockerhub`/The `dockerhub` 

For consistency and brevity?

adellape replied:
:+1:

adellape commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Document: Create an Adapter Registry in Broker to Communicate with AtomicRegistry portion of RHCC (ISV Images)](https://trello.com/c/FUsFZxkt/582-document-create-an-adapter-registry-in-broker-to-communicate-with-atomicregistry-portion-of-rhcc-isv-images)

adellape commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Document: Multiple Registry Adapters configured in one Broker Instance](https://trello.com/c/xiEEG6qi/585-document-multiple-registry-adapters-configured-in-one-broker-instance)

bmcelvee commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,599 @@
+|N
+
+|`launch_apb_on_bind`
+|Allow bind to be a no op.
```
s/no op/no-op?

adellape replied:
:+1:

bmcelvee commented:
Just a few things, otherwise LGTM! ❄️

adellape commented:
@bmcelvee Updates made per comments, thanks!

adellape commented:
Updates made per QE feedback.

adellape commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,604 @@
+registry: <1>
+  - type: dockerhub
+    name: docker
+    url: https://registry.hub.docker.com
```
I made sure `https://` prefixed all `url:` fields throughout.

adellape commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,604 @@
+    type: openshift
+    user: <user>
+    pass: <password>
+    url: https://registry.connect.redhat.com
```
I replaced the replaceable entry here w/ the URL itself.

adellape commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,604 @@
+Ansible broker] (OAB) is deployed in a cluster, its behavior is largely dictated
+by the broker's configuration file loaded on startup. The broker's configuration
+is stored as a ConfigMap object in the broker's namespace
+(*openshift-ansible-service-broker* by default).
```
Updated the default namespace here to match the `-n` options in the below commands, per QE feedback. cc @dymurray

adellape commented:
@dymurray Can you take a look at the latest commit which addresses your comments in https://bugzilla.redhat.com/show_bug.cgi?id=1535026?

adellape commented:
Oh and here's an update preview of the affected sections:

- http://file.rdu.redhat.com/~adellape/121317/brokerconfig/install_config/oab_broker_configuration.html#oab-broker-config-registry (added `auth_type` and `auth_name` to the table)
- http://file.rdu.redhat.com/~adellape/121317/brokerconfig/install_config/oab_broker_configuration.html#oab-broker-config-registry-storing-creds (new subsection)

adellape commented:
@bmcelvee Would you mind taking a look again at this one, per my latest comments?

adellape commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,673 @@
+    auth_name: asb-auth-secret
+----
+
+The associated secret should have the values `username` and `password` defined.
```
@dymurray I got this from https://github.com/openshift/ansible-service-broker/blob/master/docs/config.md#storing-registry-credentials-in-a-secretfile and want to make sure it's accurate to say `username` and `password` here for what to put in the secret, versus `user` and `pass` which is what the registry section uses in the ConfigMap.

dymurray replied:
@AlexanderZagaynov Yes `username` and `password` is correct for this.

AlexanderZagaynov replied:
@dymurray I guess you wanted to mention @adellape

adellape commented:
Updates made per QE feedback in https://bugzilla.redhat.com/show_bug.cgi?id=1535026#c3.

dymurray commented:
@AlexanderZagaynov Thank you, this looks great and I agree with commenting out `file` auth_type` for now.

AlexanderZagaynov commented:
@dymurray please pay attention on your mentions :)

dymurray commented:
@adellape Thanks looks great! I agree with commenting out `file` auth_type for now until the bug is resolved.

dymurray commented:
Sorry about the wrong mention @AlexanderZagaynov :) Not enough coffee this morning.

dymurray commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,674 @@
+----
+registry:
+  - name: rhcc
+    type: rhcc
```
@adellape Actually this should be `type` isv

adellape replied:
Oops, right!

jianzhangbjz commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,674 @@
+<1> Using the `openshift` registry type currently requires that `auth_type` be
+declared in the configuration, either to `config` or `secret`, due to a bug that
+will be addressed in a future release
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1526949[BZ#1526949]). The
```
Actually, for 3.7 version, it works fail only if specify the `auth_type: config`. But, for 3.9 version, it is not necessary. 
I think we should clarify this point and add the description of the `file` type.

jianzhangbjz commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,674 @@
+[source,yaml]
+----
+registry:
+  - name: openshift
```
The name should be `isv`.

jianzhangbjz commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,674 @@
+the `openshift` registry type, due to a bug that will be addressed in an
+upcoming release
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1526949[BZ#1526949]).
+`auth_name` is only required in this case if `auth_type` is set to `secret`. See xref:oab-broker-config-isv[ISV Registry\].]
```
The footnote should also be updated according to my comments below.

adellape commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,674 @@
+|`auth_type`
+|How the broker should read the registry credentials. Can be `config` (uses
+`user` and `pass` as defined in the `registry` section) or `secret` (uses a secret
+in the broker namespace).
```
@jianlinliu Are you referring to the instructions from the comment in https://bugzilla.redhat.com/show_bug.cgi?id=1539310#c2 ? Because I had removed the `file` type instructions until that BZ was resolved.

jianzhangbjz replied:
From your PR merge requestion, I think this doc will be released in version 3.9. If so, that bug will be fixed until then. So, I think we should add the `file` type instruction.

adellape commented:
Fixed the isv/openshift issue, and squashed down to one commit. 

@jianzhangbjz Waiting for confirmation on if I should include the `file` type or not, per https://github.com/openshift/openshift-docs/pull/6755#discussion_r165128162.

adellape commented:
Given the resolution of https://bugzilla.redhat.com/show_bug.cgi?id=1539310#c3 about the `file` type issue, I've added the `file` type back into the doc here, and also expanded the instructions on how to use the `secret` or `file` type:

http://file.rdu.redhat.com/~adellape/121317/brokerconfig/install_config/oab_broker_configuration.html#oab-config-registry-storing-creds

@jianlinliu @dymurray PTAL again.

adellape commented:
@jianlinliu And as for 3.7 vs 3.9, I want to merge this to master and cherry-pick it to the 3.7 docs immediately, then I will make a follow-up PR to remove the `config` requirement and that will be for 3.9 docs.

jianlinliu commented:
@jianzhangbjz pls help take care of the above comments.

jianzhangbjz commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,732 @@
+registry:
+  - name: rhcc
+    type: rhcc
+    url: registry.access.redhat.com
```
I think the `registry.connect.redhat.com` would be better.

jianzhangbjz commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,732 @@
+
+----
+registry:
+  - name: openshift
```
This name should be `isv`.

jianzhangbjz replied:
Also should add this part: `url: registry.connect.redhat.com`

dymurray replied:
+1 url should be prefixed with `https` too.

adellape replied:
:+1:

jianzhangbjz commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,732 @@
+*_/tmp/registry-credentials/reg-creds.yaml_*:
++
+----
+$ oc edit asb -n openshift-ansible-service-broker
```
This command should be `oc edit dc/asb -n openshift-ansible-service-broker`.

adellape replied:
Whoops! Thanks.

jianzhangbjz commented:
@jianlinliu Thanks for your information. @adellape Thanks for your update! I added some comments, please help to update them.

adellape commented:
Great, thank you @jianzhangbjz @dymurray. Fixed those mistakes and squashed to one commit.

adellape commented:
[rev_history]
|xref:../install_config/oab_broker_configuration.adoc#install-config-oab-config[OpenShift Ansible Broker Configuration]
|New topic describing the OAB broker configuration file and how to update it.
%

jianzhangbjz commented on install_config/oab_broker_configuration.adoc:
```diff
@@ -0,0 +1,733 @@
+that will be addressed in a future release
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1526949[BZ#1526949]). See
+xref:oab-config-registry-storing-creds[Storing Registry Credentials] for
+options.
```
For 3.9 version, here instructions are no need. For 3.7, it's well.
